add `<Icon type="status-icon" status={success,error,warning} />`

stephenmathieson opened the pull request:
@anthonyshort @dominicbarnes @stevenmiller888

stephenmathieson committed:
lib: add `error`, `success` and `warning`

stephenmathieson committed:
error,success,warning: remove xml stuff

stephenmathieson committed:
error,success,warning: remove more xml stuff

anthonyshort commented on lib/icons/success.js:
```diff
@@ -0,0 +1,26 @@
+
```
OCD: Extra space :D

stephenmathieson replied:
makes editing in vim significantly easier ;)

anthonyshort commented on lib/icons/error.js:
```diff
@@ -0,0 +1,22 @@
+export default function render(color) {
+  return (
+    <svg width="16px" height="16px" viewBox="0 0 16 16">
+      <title>error</title>
```
You can actually remove the title, desc, defs elements from all of these

stephenmathieson replied:
cool, will do

anthonyshort commented:
https://segment.atlassian.net/browse/WARE-54

I've attached some updated icons in jira. There were some extra elements and groups in the ones I originally sent. They will probably mess up Firefox and Safari if we don't update them. 

Other than that, it's all good to me.

stephenmathieson commented:
thought about abstracting this a bit further:

``` js
import StatusIcon from 'segmentio/icons';

// ...
<StatusIcon type="success|warning|error" />
```

thoughts?

anthonyshort commented:
Yeah I'd be down for getting these 3 icons to be a separate thing. They're used in the alert boxes too (but bigger). Since they're not the same style as the other icons, it could be good to have them separate instead.

So +1!

stephenmathieson commented:
~~cool, i'll update for `StatusIcon`~~

done

stephenmathieson committed:
ignore: keep `./duo` directory out of git

stephenmathieson committed:
lib: remove "error", "success" and "warning"

These will be re-added as a seperate `StatusIcon` component.

stephenmathieson committed:
lib: add `StatusIcon`

This component will take the place of the prior `Error`, `Success` and
`Warning` icons.

To use, do `<Icon type="success-icon" status="success|warning|error" />`

stephenmathieson committed:
preview: attempt to get `status-icon` working

This is throwing `Uncaught ReferenceError: dom is not defined` and
needs some work still...

Only pushing so @stevenmiller888 can help be debug.

stephenmathieson committed:
status-icon: add "@jsx element" directive

No idea what this thing does, but it's apparently required.  We
desperately need documentation here.

This JSX stuff just went from "this is pretty cool" to "this is a super
leaky abstraction and it's not worth it" in my brain.

stephenmathieson committed:
preview: change "dom" -> "element"

stephenmathieson committed:
status-icon: remove "@jsx element" directives

stephenmathieson committed:
preview: wrap each `StatusIcon` in its own `<div />`

stevenmiller888 commented:
LGTM

stephenmathieson committed:
readme: add "status-icon" to list
